In [24]:
# First run src/data_sources/get_data.py to get the csv in your working directory.
import pandas as pd

companies_df = pd.read_csv('game_companies.csv')
companies_df.head()

Unnamed: 0  id                     name    start_date  changed_company_id   
0           0   1          Electronic Arts  3.913920e+08                 5.0  \
1           1   2                  BioWare  7.915968e+08                 NaN   
2           2   4        Eidos Interactive  8.335008e+08              4940.0   
3           3   5  Interplay Entertainment  4.416768e+08                 NaN   
4           4   8                 2K Games  1.106611e+09                 NaN   

   parent   change_date  
0     NaN           NaN  
1     1.0           NaN  
2    26.0  1.257725e+09  
3     NaN           NaN  
4   139.0           NaN

In [25]:
# Check the data types of the columns.
companies_df.dtypes

Unnamed: 0              int64
id                      int64
name                   object
start_date            float64
changed_company_id    float64
parent                float64
change_date           float64
dtype: object

In [26]:
# Check how many unique companies were returned.
companies_df['id'].nunique()

12659

We see that there are 12659 unique companies that have either published or developed a video game since 1 October 2000.

In [27]:
# Check how many companies do not have data for start_date.
companies_df[companies_df.start_date.notnull()].shape[0]

1946

There are 1946 companies for which the founding date of the company is known.

In [32]:
# Filter out companies with a parent as the parent will be publicly listed.
companies_no_child_df = companies_df[~companies_df.parent.notnull()]
companies_no_child_df.head()
companies_no_child_df.shape[0]

12166

In [52]:
# Get a dataframe of parent companies.
parent_ids = companies_df.parent.dropna()
print(parent_ids.shape[0])
parent_ids.head()

493


1        1.0
2       26.0
4      139.0
5    13016.0
8      245.0
Name: parent, dtype: float64

493 parent companies. These are the most likely to be publicly traded.

In [55]:
companies_parents_df = companies_df[companies_df.id.isin(parent_ids)]
companies_parents_df.head()

Unnamed: 0  id             name    start_date  changed_company_id  parent   
0            0   1  Electronic Arts  3.913920e+08                 5.0     NaN  \
1            1   2          BioWare  7.915968e+08                 NaN     1.0   
4            4   8         2K Games  1.106611e+09                 NaN   139.0   
11          11  26      Square Enix  1.049155e+09                 NaN     NaN   
17          17  41       Riot Games  1.156982e+09                 NaN  3595.0   

    change_date  
0           NaN  
1           NaN  
4           NaN  
11          NaN  
17          NaN